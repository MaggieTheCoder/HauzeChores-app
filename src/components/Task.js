import React from "react";
import propTypes from "prop-types";
import deleteTaskById from "../requests/deleteTaskById";
import assignTaskToUser from "../requests/assignTaskToUser";
import "../styles/Task.css";

const Task = ({ taskname, id, setAddedANewTask, userId }) => {
  return (
    <div className="task">
      <button
        type="button"
        className="task__button"
        onClick={() => {
          assignTaskToUser(userId, id, setAddedANewTask);
        }}
      >
        Take it!
      </button>
      <p className="task__name">{taskname}</p>
      <button
        type="button"
        onClick={() => deleteTaskById(id, setAddedANewTask)}
        className="task__button"
      >
        X
      </button>
    </div>
  );
};

export default Task;

Task.propTypes = {
  taskname: propTypes.string.isRequired,
  id: propTypes.number.isRequired,
  setAddedANewTask: propTypes.func.isRequired,
  userId: propTypes.number.isRequired,
};
