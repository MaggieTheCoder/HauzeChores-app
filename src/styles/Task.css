.task {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 10px;
  font-size: 20px;
  margin-left: 10px;
  padding-left: 100px;
  padding-right: 100px;
  scroll-behavior: auto;
}

.task__button {
  font-family: "Menlo", sans-serif;
  font-size: 25px;
  background-color: #C60982;
  padding: 3px 50px 3px 50px;
  border-radius: 15px;
  color: white;
}

.task__name {
  color: #C60982;
  font-size: 25px;
}
