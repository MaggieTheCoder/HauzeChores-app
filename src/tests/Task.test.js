import React from "react";
import { render } from "@testing-library/react";
import Task from "../components/Task";

test("renders Task component correctly", () => {
  const { asFragment } = render(<Task />);

  expect(asFragment()).toMatchSnapshot();
});
