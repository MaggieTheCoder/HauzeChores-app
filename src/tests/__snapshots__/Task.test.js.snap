// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Task component correctly 1`] = `
<DocumentFragment>
  <div
    class="task"
  >
    <button
      class="task__button"
      type="button"
    >
      Take it!
    </button>
    <button
      class="task__button"
      type="button"
    >
      X
    </button>
    <p
      class="task__name"
    />
  </div>
</DocumentFragment>
`;
